spec/models/spree/gateway/amazon_spec.rb

```diff
@@ -14,20 +25,19 @@
 
       expect(mws).to have_received(:refund).with("CAPTURE_ID", payment.number, 30.0, "USD")
     end
-  end
 
-  describe '#authorize' do
-    let(:payment_method) { Spree::Gateway::Amazon.create!(name: 'Amazon', preferred_test_mode: true) }
-    let(:order) { create(:order_with_line_items, state: 'delivery') }
-    let(:payment_source) { Spree::AmazonTransaction.create!(order_id: order.id, order_reference: 'REFERENCE') }
-    let!(:payment) do
-      create(:payment,
-             order: order,
-             payment_method: payment_method,
-             source: payment_source,
-             amount: order.total)
+    let!(:refund) { create(:refund, payment: payment, amount: payment.amount) }
+    it 'succeeds' do
+      response = build_mws_refund_response(state: 'Pending', total: order.total)
+      expect(payment_method.send(:load_amazon_mws, 'REFERENCE')).to receive(:refund).and_return(response)
```

Comment:
Not your fault, but just noticed this `load_amazon_mws`. 

1. The correct way is to stub the class instance it represents:
```ruby
AmazonMws.new(reference, self.preferred_test_mode)
```
```
allow(AmazonAws.to receive(:new).with(....).and_return(..)
```

Because your code's dependency is not this private function, but this external class.

2. You can also think about extracting it into a helper function in case the only change is the `receive` parameter
```ruby
def stub_amazon_response!(action)
  allow(AmazonAws.to receive(action)....)
end
```